src/MatchingResult.php

```diff
@@ -9,22 +9,18 @@
 
 final class MatchingResult implements MiddlewareInterface
 {
-    private $success;
-
-    /**
-     * @var Route
-     */
-    private $route;
-    private $parameters = [];
-    private $methods;
+    private bool $success;
+    private Route $route;
+    private array $parameters = [];
+    private array $methods = Method::ANY;
 
     private function __construct()
     {
     }
 
     public static function fromSuccess(Route $route, array $parameters): self
     {
-        $new = new self();
+        $new = new static();
```

Comment:
Please have an attension here. Is this ok ot there was some reason to leave `self`?

Reply:
The class is final so `static` doesn't make sense.

Reply:
Made `self` in all cases here